lib/probes/MemoryProbe.js

```diff
@@ -0,0 +1,37 @@
+var Q = require('q'),
+	util = require('util'),
+	events = require('events'),
+	helpers = require('../helpers'),
+	debug = require('debug')('bp:probes:MemoryProbe');
+
+function MemoryProbe(id) {
+	if (id) {
+		this.id = id;
+	}
+	events.EventEmitter.call(this);
+}
+
+util.inherits(MemoryProbe, events.EventEmitter);
+
+MemoryProbe.prototype.id = 'MemoryProbe';
+
+MemoryProbe.prototype.setup = function(cfg) {
+	this.timerHandle = null;
+};
+
+MemoryProbe.prototype.start = function(browser) {
+	var me = this;
+	this.timerHandle = setInterval(function() {
+		browser.eval('window.performance.memory').then(function(res) {
+			if(res) {
+				me.emit('data', res);
```

Comment:
I am noticing some inconsistencies when the window gets scrolled (browser scripts seem to freeze). I suspect it is related to this code. Do you have any pointers? As you said, storing data browser-side would impact the results. An alternative would be to store max, min and last values browser-side. It would have minimum impact. Or remove the timer entirely and just return the last values before the test stops. Any other options?